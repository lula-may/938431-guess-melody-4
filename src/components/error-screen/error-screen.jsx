import React from "react";
import PropTypes from "prop-types";

const ErrorScreen = (props) => {
  const {errorMessage} = props;
  return (
    <section className="result">
      <div className="result__logo"><img src="img/melody-logo.png" alt="Угадай мелодию" width="186" height="83"/></div>
      <h2 className="result__title" style={{color: `red`}}>Ошибка загрузки данных.</h2>
      <p className="result__total result__total--fail">
    Упсс... У нас что-то сломалось. {errorMessage}
      </p>
    </section>
  );
};

ErrorScreen.propTypes = {
  errorMessage: PropTypes.string.isRequired,
};

export default ErrorScreen;
