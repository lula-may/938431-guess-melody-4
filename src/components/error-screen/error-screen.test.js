import React from "react";
import renderer from "react-test-renderer";
import "./error-screen.jsx";
import ErrorScreen from "./error-screen.jsx";

describe(`ErrorScreen Component`, () => {
  it(`should correctly render ErrorScreen`, () => {
    const tree = renderer.create(
        <ErrorScreen/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
