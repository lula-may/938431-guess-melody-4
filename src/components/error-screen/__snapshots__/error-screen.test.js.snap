// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorScreen Component should correctly render ErrorScreen 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
    style={
      Object {
        "color": "red",
      }
    }
  >
    Ошибка загрузки данных.
  </h2>
  <p
    className="result__total result__total--fail"
  >
    Упсс... У нас что-то сломалось. Попробуйте зайти позднее.
  </p>
</section>
`;
